File: gluecodium/src/test/resources/smoke/nesting/output/cpp/src/smoke/LevelOne.cpp
// -------------------------------------------------------------------------------------------------
//
//
// -------------------------------------------------------------------------------------------------
#include "smoke/LevelOne.h"
#include <utility>
namespace smoke {
LevelOne::LevelOne() {
}
LevelOne::~LevelOne() {
}
LevelOne::LevelTwo::LevelTwo() {
}
LevelOne::LevelTwo::~LevelTwo() {
}
LevelOne::LevelTwo::LevelThree::LevelThree() {
}
LevelOne::LevelTwo::LevelThree::~LevelThree() {
}
const bool LevelOne::LevelTwo::LevelThree::LevelFour::FOO = false;
LevelOne::LevelTwo::LevelThree::LevelFour::LevelFour( )
    : string_field{ }
{
}
LevelOne::LevelTwo::LevelThree::LevelFour::LevelFour( ::std::string string_field )
    : string_field( std::move( string_field ) )
{
}
}

File: gluecodium/src/test/resources/smoke/nesting/output/android-kotlin/com/example/smoke/LevelOne.kt
/*

 *
 */

@file:JvmName("LevelOne")

package com.example.smoke

import com.example.NativeBase

class LevelOne : NativeBase {

    class LevelTwo : NativeBase {

        class LevelThree : NativeBase {

            enum class LevelFourEnum(private val value: Int) {
                NONE(0);
            }
            class LevelFour {
                @JvmField var stringField: String



                constructor(stringField: String) {
                    this.stringField = stringField
                }





                companion object {
                    @JvmField final val FOO: Boolean = false
                    @JvmStatic external fun fooFactory() : LevelOne.LevelTwo.LevelThree.LevelFour
                }
            }



            /*
             * For internal use only.
             * @hidden
             * @param nativeHandle The handle to resources on C++ side.
             * @param tag Tag used by callers to avoid overload resolution problems.
             */
            protected constructor(nativeHandle: Long, tag: Any?)
                : super(nativeHandle, { disposeNativeHandle(it) }) {}



            external fun foo(input: OuterClass.InnerInterface) : OuterInterface.InnerClass




            companion object {
                @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
            }
        }


        /*
         * For internal use only.
         * @hidden
         * @param nativeHandle The handle to resources on C++ side.
         * @param tag Tag used by callers to avoid overload resolution problems.
         */
        protected constructor(nativeHandle: Long, tag: Any?)
            : super(nativeHandle, { disposeNativeHandle(it) }) {}







        companion object {
            @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
        }
    }


    /*
     * For internal use only.
     * @hidden
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
